docs/src/lisp/bindings_scopes.md

```diff
@@ -0,0 +1,81 @@
+# Bindings and Scopes
+
+How do we define symbols? And more importantly, how do we assign values to them? Let's address these
+questions now.
+
+```lisp
+λ > a
+```
+
+Congratulations, we've defined our first symbol.
+
+```
+error[E02]: FreeIdentifier
+  ┌─ :1:1
+  │
+1 │ a
+  │ ^ a
+```
+
+Well, not so fast.
+
+As mentioned earlier in the book, the interpreter evaluates S-expressions that it encounters, and
+the symbol `a` is no exception. Since symbol evaluation involves substituting it with a bound value,
+and there is none bound to it, we get an error.
+
+Let's change that.
+
+### `define`
+
+This is how we bind values to symbols, using `(define symbol value)`:
+
+```
+λ > (define a 1337)
+λ > a
+=> 1337
+```
+
+Here, we've created a _binding_ between the symbol `a` and a number literal. Consequent ly,
+further evaluation of `a` yields the elite integer. It's worth noting that regardless of whether
+`a` is bound and used across different expression s (in separate interprete r entries), it still
+retains its value. Why? Because the Steel session we're in introduces a _scope_, a context used in
+evaluation to resolve symbol bindings.
+
+Another example:
+
+```
+λ > (define c (+ 1000 337))
+λ > c
+=> 1337
+```
+
+Although evaluation rules differ for `define` (symbol `c` is not evaluated, otherwise it would
+result in an error and make no sense), the _value_ part follows standard rules.
+
+### `let`
+
+If `define` seems reminiscent of global variables to you, you're not alone. While it's suitable for
+defining function bindings (more on that later) or constants, there must be a better way for local
+bindings. Allow me to introduce `let`, a term familiar from our host language, Rust, with the same
+meaning.
+
+`let` is used as follows: `(let ((symbol-a value-a) (symbol-b value-b) ...) body)`.
+
+```lisp
+λ > (let ((a (* 25 4))
+          (b 3))
+      (* a b))
+=> 300
+λ > a
+=> 1337
+```
```

Comment:
In steel, `define` desugars to `let` when used in a local scope, so I think this distinction should be made here. In fact, I'd argue that define is preferred to let stylistically

Reply:
Could you please provide an example? I'm not so familiar with steel and learned it as I wrote this, I have only elisp and clojure experience to rely on

Reply:
I tried to use `define` in a local scope as far as I understood it correctly: 
```
λ > (let ((a 5)) (define b 4) (+ a b))
=> 9
λ > b
=> 4
```

And it escaped; was it a local scope you mentioned and this is a bug?

Reply:
That is a bug! Good catch, look like this is a regression and I missed a test. I'll get a fix up

What I was referring to is like this:

```scheme
(define (test) (define foo 10) (+ foo 100))
```

Reply:
ah, its a bug with const evaluation. See that this works fine:

```scheme
(let ((foo (list 10 20 30 40))) (define bar (list 50 60)) (append foo bar))
```

Reply:
Will get patched by this #200

Reply:
I've updated the chapter with both ways of defining symbols locally, please take a look